packs/chef/actions/lib/omnibus.py

```diff
@@ -1,40 +1,51 @@
+import pipes
+
 import shellhelpers as shell
 
+
 class Omnibus(object):
     OMNITRUCK = 'https://www.chef.io/chef/install.sh'
 
     def __init__(self, options):
         supports = ('pre_release', 'version', 'download_path')
-        self.options = { k: options[k] for k in supports }
+        self.options = {k: options[k] for k in supports}
 
-    def build_command(self, version=None, pre_release=False, download_path=None):
-        command = "curl -sL %s | sudo bash" % self.OMNITRUCK
+    def build_command(self, version=None, pre_release=False,
+                      download_path=None):
+        command = "curl -sL %s | sudo bash" % (pipes.quote(self.OMNITRUCK))
 
-        if pre_release or version != None or download_path != None:
-          command += " -s --"
+        command_args = []
+
+        if pre_release or version or download_path:
+            command_args += ['-s', '--']
 
         if pre_release:
-          command += " -p"
+            command_args += ['-p']
 
-        if version != None:
-          command += " -v \"%s\"" % version
+        if version:
+            command_args += ['-v', version]
 
         if download_path:
-          command += " -d \"%s\"" % download_path
+            command_args += ['-v', download_path]
 
-        return command
+        command_args = [pipes.quote(arg) for arg in command_args]
+        command_args = ' '.join(command_args)
+        command = command + ' ' + command_args
 
+        return command
 
     def chef_installed(self):
         knife = "/opt/chef/bin/knife"
-        command = "test -x %s" % knife
+        command = "test -x %s" % (pipes.quote(knife))
```

Comment:
I agree about security and thank you for the point. Also cool to know that it will work in Python 3.3.
But we should keep in mind also  what is crucial to escape and what is not. Regarding that the path configurable or not, the user has already remote access via ssh isn't he? And even more full sudo privileges, thus consequently the "securing" code doesn't make the whole solution anyhow more secure...